### PR TITLE
feat: Implement latest blog endpoint

### PR DESCRIPTION
**Pull request by JumongComputers**
## What does the PR do?

This PR is about fetching latest blog post for users when not logged in . The endpoint returns errors if:

- [x] there is no blog post created.

How should this be manually tested?

1. Make a GET request to the `/api/blogs/latest` endpoint with the following payload:

Successful Response

On a successful blogs latest response, the API should return a 200 OK status code.

```json
{
        "count": 5,
        "next": "http://example.com/api/blogs/latest?page=2&page_size=10",
        "previous": null,
        "results":
        [{
            "title": "Latest Blog Post Title",
            "excerpt": "Short excerpt of the blog post...",
            "publish_date": "2024-07-17T00:00:00Z",
            "author": "Author Name"
        }]
}
```

Error Response:

Status: 500 Internal Server Error

Body:
```json
{
    "error": "Internal server error."
}
```
Invalid Method Response:

Status: 405 Method Not Allowed

Body:

On Invalid error, if the user is not found with the provided id:
```json
    {
        "error": "This method is not allowed."
    }
```

Bad Request Response:

Status: 400 Bad Request

Body:
```json
    {
        "error": "An invalid request was sent."
    }
```

Task

- [x] Create a GET endpoint  /api/v1/blogs/latest to create testimonials for logged in users.
- [x] Implement server-side logic to handle successful and failed response.
- [x] Write tests for fetching latest blog post.
- [x] Update API documentation to reflect the new endpoint and its usage.

Reference Issue: [https://github.com/hngprojects/hng_boilerplate_nestjs/issues/134)

### CODE REVIEW

**Comment by davidolanrewaju** [comment]
Fix the merge conflicts

File: package.json
Comment (Heba-WebDev) [inline package.json:1]: Why did you change bcrypt to bcryptjs? This is not a personal project, you can't change one dependency with another by yourself

**Review by Heba-WebDev — CHANGES_REQUESTED** [review]
I'll approve once the conflicts are solved

**Review by Heba-WebDev — DISMISSED** [review]
Looks good to me but I'd need @Homoakin619 to take a look as well

**Comment by JumongComputers** [comment]
@Homoakin619 I have resolved the conflict

**Pull request closed without merging**